crates/oxc_minifier/src/peephole/remove_unused_expression.rs

```diff
@@ -626,6 +650,31 @@ impl<'a> PeepholeOptimizations {
         false
     }
 
+    fn fold_expression_in_mathematical_operation(
+        &self,
+        e: &mut Expression<'a>,
+        state: &mut State,
+        ctx: Ctx<'a, '_>,
+    ) -> bool {
+        match e {
+            Expression::UnaryExpression(unary_expr) => {
+                if unary_expr.operator.is_arithmetic() {
+                    *e = unary_expr.argument.take_in(ctx.ast.allocator);
+                    state.changed = true;
+                }
+            }
+            Expression::StringLiteral(literal) => {
+                if literal.value.is_empty() {
+                    *e = ctx.ast.number_0();
+                    state.changed = true;
+                    return true;
+                }
+            }
```

Comment:
```suggestion
            Expression::StringLiteral(literal) => {
                if literal.value.is_empty() {
                    // NOTE: this changes the result value, but it's fine because the return value is not used
                    // (example case: `"" > "-1"`)
                    *e = ctx.ast.number_0();
                    state.changed = true;
                    return true;
                }
            }
```
Let's add a comment to make it clear that this changes the result value.